core/src/test/scala/unit/kafka/server/GroupCoordinatorBaseRequestTest.scala

```diff
@@ -181,6 +184,52 @@ class GroupCoordinatorBaseRequestTest(cluster: ClusterInstance) {
     response.data.groups.asScala.toList
   }
 
+  protected def deleteOffset(
+    groupId: String,
+    topic: String,
+    partition: Int,
+    expectedResponseError: Errors = Errors.NONE,
+    expectedPartitionError: Errors = Errors.NONE,
+    version: Short
+  ): Unit = {
+    if (expectedResponseError != Errors.NONE && expectedPartitionError != Errors.NONE) {
+      fail("deleteOffset: neither expectedResponseError nor expectedTopicError is Errors.NONE.")
+    }
+
+    val request = new OffsetDeleteRequest.Builder(
+      new OffsetDeleteRequestData()
+        .setGroupId(groupId)
+        .setTopics(new OffsetDeleteRequestData.OffsetDeleteRequestTopicCollection(List(
+          new OffsetDeleteRequestData.OffsetDeleteRequestTopic()
+            .setName(topic)
+            .setPartitions(List(
+              new OffsetDeleteRequestData.OffsetDeleteRequestPartition()
+                .setPartitionIndex(partition)
+            ).asJava)
+        ).asJava.iterator()))
+    ).build(version)
+
+    val expectedResponse = new OffsetDeleteResponseData()
+    if (expectedResponseError == Errors.NONE) {
+      expectedResponse
+        .setTopics(new OffsetDeleteResponseData.OffsetDeleteResponseTopicCollection(List(
+          new OffsetDeleteResponseData.OffsetDeleteResponseTopic()
+            .setName(topic)
+            .setPartitions(new OffsetDeleteResponseData.OffsetDeleteResponsePartitionCollection(List(
+              new OffsetDeleteResponseData.OffsetDeleteResponsePartition()
+                .setPartitionIndex(partition)
+                .setErrorCode(expectedPartitionError.code())
+            ).asJava.iterator()))
+        ).asJava.iterator()))
+    } else {
+      expectedResponse
+        .setErrorCode(expectedResponseError.code())
```

Comment:
nit: We could bring back `setErrorCode` on the previous line as there is space for it.